li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -349,4 +372,28 @@ public void close(Duration timeout) {
   public void wakeup() {
     throw new UnsupportedOperationException("Not implemented yet");
   }
+
+  // Intended for testing only
+  LiKafkaConsumer<K, V> getPerClusterConsumer(ClusterDescriptor cluster) {
+    return _consumers.get(cluster);
+  }
+
+  private LiKafkaConsumer<K, V> getOrCreateConsumerForTopic(String topic) {
+    return getOrCreatePerClusterConsumer(_mdsClient.getClusterForTopic(_clientId, topic, _mdsRequestTimeoutMs));
```

Comment:
i think you should handle clusterForTopic() returning null

Reply:
Added a null check to getOrCreatePerClusterConsumer().